File: NgheAn_Vinh_2324/Cau1/CAYXANH.py
# Đọc dữ liệu từ file CAYXANH.INP
with open('CAYXANH.INP', 'r') as file:
    # Đọc dòng đầu tiên và tách các giá trị n, l, m
    n, l, m = map(int, file.readline().split())
    
    # Đọc dòng thứ hai và tách thành danh sách các chiều cao cây
    heights = list(map(int, file.readline().split()))

# Khởi tạo biến đếm
count = 0

# Duyệt qua từng chiều cao cây và kiểm tra điều kiện
for height in heights:
    if abs(height - l) <= m:
        count += 1

# Ghi kết quả vào file CAYXANH.OUT
with open('CAYXANH.OUT', 'w') as file:
    file.write(str(count))

File: THCS/2023_2024/NgheAn_Vinh_2324/Cau1/CAYXANH.cpp
#include <iostream>
#include <fstream>
#include <cmath>
using namespace std;

int main() {
    // Mở file CAYXANH.INP để đọc dữ liệu
    ifstream input("CAYXANH.INP");
    ofstream output("CAYXANH.OUT");

    // Đọc n, l, m
    int n;
    long long l, m;
    input >> n >> l >> m;

    // Khởi tạo biến đếm số cây thỏa mãn điều kiện
    int count = 0;
    long long height;

    // Duyệt qua từng chiều cao cây
    for (int i = 0; i < n; i++) {
        input >> height;
        if (abs(height - l) <= m) {
            count++;
        }
    }

    // Ghi kết quả ra file CAYXANH.OUT
    output << count;

    // Đóng các file
    input.close();
    output.close();

    return 0;
}
